packages/safe-ds-vscode/src/extension/messages.ts

```diff
@@ -68,7 +68,27 @@ export interface ProgramMainInformation {
 export interface PlaceholderQueryMessage {
     type: 'placeholder_query';
     id: string;
-    data: string;
+    data: PlaceholderQuery;
+}
+
+/**
+ * A query on a placeholder value.
+ */
+export interface PlaceholderQuery {
+    /**
+     * The name of the requested placeholder.
+     */
+    name: string;
+
+    /**
+     * The offset of the requested data.
+     */
+    window_begin?: number;
+
+    /**
+     * The size of the requested data.
+     */
+    window_size?: number;
```

Comment:
Is it intended that you can also specify only `window_begin` or `window_size`? What's the behavior in this case? I guess `window_begin` defaults to the start of the data and `window_size` to the size of the remaining data?

Reply:
That is correct